### PR TITLE
Make use of sub-queries, to get rid of most apoc calls.

### LINKED ISSUE #214: Try to replace apoc calls by subqueries

### PR DESCRIPTION
**Pull request by Andy2003**
resolves #214

### CODE REVIEW

**Comment by Andy2003** [comment]
only thing left is updating to ne next release of the neo4j-cypher-dsl

**Comment by michael-simons** [comment]
> only thing left is updating to ne next release of the neo4j-cypher-dsl

Feel free to ping me if needed. 
Next week after hack/commit/push is totally fine.

File: core/src/test/resources/custom-fields.adoc [hunk 1]
```diff
@@ -86,10 +86,10 @@ mutation { createPerson(name:"Jill") {name} }
 .Cypher
 [source,cypher]
 ----
-CALL apoc.cypher.doIt('WITH $name AS name CREATE (p:Person { name: \"Test\"+$name }) RETURN p',  {
-	name: $createPersonName
-}) YIELD value
-WITH value[head(keys(value))] AS createPerson
+CALL {
+	WITH $createPersonName AS name
```
Comment (jexp) [inline core/src/test/resources/custom-fields.adoc:90]: we need to watch out here for naming conflicts if we rename to a single/simple name, we should still be able to use parameters in the subqueries

File: core/src/test/resources/custom-fields.adoc [hunk 2]
```diff
@@ -169,14 +170,22 @@ query { Person(name:"Jane") {name, label} }
...
 [source,cypher]
 ----
-UNWIND apoc.cypher.runFirstColumnSingle('WITH $name AS name MATCH (p:Person { name: $name }) RETURN p',  {
-	name: $personName
-}) AS person
+CALL {
+	WITH $personName AS name
+	MATCH (p:Person { name: name }) RETURN p AS person LIMIT 1
+}
+CALL {
```
Comment (jexp) [inline core/src/test/resources/custom-fields.adoc:177]: why did we have the 2 subqueries here if the 2nd outer just is a pass-through, can we detect that and remove/skip it if it has no actual operations?

File: core/src/test/resources/custom-fields.adoc (hunk 1 above)
Comment (Andy2003) [reply core/src/test/resources/custom-fields.adoc:90]: I see no potential conflict here:

- `$createPersonName` is defined by the library
- `name` is the name of the fields argument defined by the user

**Review by jexp — APPROVED** [review]
Looks great, thanks so much. Sorry it took so long.

**Pull request merged**